widgets/common/src/filters/RulesForm.tsx

```
@@ -46,7 +46,7 @@ function getFilterRule(filterRuleRow: FilterRuleRow): FilterRule {
     return filterRuleRow.rule;
 }
 
-interface RulesFormProps {
+export interface RulesFormProps {
```

I checked removing that and type-checking called from webpack devserver did not complain about it, but when I run `npm run check-types` I got errors (the same goes to `LabelsInputProps` and `ValidationErrorPopupProps`).

Is it expected that we will get different output from webpack with type-checking and check-types script?

---

It turns out it was not the case. This error occurred in `npm run check-types` despite having `declarationMap: false`. My other hunch is that declaration errors are not emitted by default due to `declaration: false` in `diagnosticOptions` in https://github.com/TypeStrong/fork-ts-checker-webpack-plugin/blob/5b7c333d91f3eed9fb134ba9b9ed10f52115e7f6/README.md#typescript-options.

I tried setting it to `true` in `webpack.config.json`, but the errors still were not reported by webpack.

It looks like only declaration errors are not reported, as other errors were reported normally, as I have seen. We can always remove `ForkTsCheckerWebpackPlugin` and run `tsc --build --watch` ourselves. It will always report all the errors, and it may be faster (just a thought) :smile: 

I think the issue is rather minor and I don't mind not seeing rare errors in webpack, so I'd rather move on. Let me know if you have other expectations :slightly_smiling_face: 

---

When experimenting, I created a shared `tsconfig.base.json` for the stage repo, which IMO is a good idea, as it reduces the number of lines of code in the configs. I will commit the changes

---

Thanks for checking. OK, let's leave it.

Adding shared config - good idea 👍🏻